src/net.h

```diff
@@ -638,14 +638,28 @@ class CNetMessage {
  */
 class TransportDeserializer {
 public:
-    // returns true if the current deserialization is complete
-    virtual bool Complete() const = 0;
     // set the serialization context version
     virtual void SetVersion(int version) = 0;
-    // read and deserialize data
-    virtual int Read(const char *data, unsigned int bytes) = 0;
-    // decomposes a message from the context
-    virtual CNetMessage GetMessage(const CMessageHeader::MessageStartChars& message_start, int64_t time) = 0;
+    /**
+     * Try to decode all CNetMessage%s from the first \p num_bytes bytes of the
+     * \p receive_buffer buffer.
+     *
+     * @param[in] message_start The expected message start string.
+     * @param[in] time The approximate time at which these messages were
+     *                 received in microseconds since UNIX epoch.
+     *
+     * @returns A tuple where the first element indicates whether or not an
+     *          unhandleable error has occured (e.g. failure to deserialize the
+     *          header or ludicrous message sizes), and the second element
+     *          contains all succesfully decoded CNetMessage%s.
+     *
+     * @note Even if an unhandleable error has occured, CNetMessage%s might have
+     *       already been decoded. These CNetMessage%s should not be ignored and
+     *       probably should be processed.
+     *
+     * @see CNode::ReceiveMsgBytes(const char *, unsigned int, bool&)
+     */
+    virtual std::tuple<bool, std::vector<CNetMessage>> ReadMessages(const char *receive_buffer, unsigned int num_bytes, const CMessageHeader::MessageStartChars& message_start, int64_t time) = 0;
```

Comment:
Any reason to use an `std::tuple` with just two members here (instead of an `std::pair`)?